.github/workflows/tests.yml

```diff
@@ -69,22 +69,21 @@ jobs:
   linux-tests:
     name: Linux Test
     runs-on: ubuntu-22.04
-    strategy:
-      matrix:
-        target: [
-          x86_64-unknown-linux-musl,
-          i686-unknown-linux-gnu,
-          i686-unknown-linux-musl,
-        ]
     steps:
       - uses: actions/checkout@v3
       - uses: dtolnay/rust-toolchain@stable
         with:
-          targets: ${{ matrix.target }}
-      - name: Install multilib
-        run: sudo apt-get update && sudo apt-get install gcc-multilib
+          targets: x86_64-unknown-linux-musl, i686-unknown-linux-gnu, i686-unknown-linux-musl, x86_64-unknown-linux-gnux32
```

Comment:
I do not like replacing multiple separate jobs with a single job that mixes targets. I think we should keep the matrix like before, and add a variable `packages` to each specific target:
```
        include:
          - target: x86_64-unknown-linux-gnux32
            cargo_test_options: --no-run --features=rdrand
            packages: "libc6-dev-x32 libx32gcc-11-dev"
...

- if ${{ matrix.packages != "" }}
  run sudo apt-get install ${{ matrix.packages }}
...
- cargo test ${{ matrix.cargo_test_options }} --features=std
```

Reply:
Nice idea, this setup also makes it clear that only the `gnu` targets need these libs installed (which makes perfect sense).

Should be fixed now, let me know what you think.